.luacheckrc

```diff
@@ -4,9 +4,9 @@ codes=true
 -- skip showing files with no issues
 quiet=1
 -- skip showing undefined variable usage
--- there are thousands of warnings here because luacheck is unaware of Wesnoth's
+-- there can be warnings because luacheck is unaware of Wesnoth's
 -- lua environment and has no way to check which have been loaded
-global=false
+globals={"wesnoth","wml","filesystem","start_x","start_y","delta_x","delta_y","index","helper"}
```

Comment:
this list contains wrong entries. start_x is not provided by wesnoth for example

Reply:
I had to add all of these to get the check to pass

Reply:
https://luacheck.readthedocs.io/en/stable/warnings.html allow_defined  sounds like it could solve code style related 111 and 113

Reply:
incorrectly passing check is worse than failing check

Reply:
based on emmylua wesnoth provides these globals   "Lua.diagnostics.globals": [
        "wesnoth",
        "wml",
        "gui",
        "filesystem",
        "unit_test",
        "stringx",
        "mathx",
        "ai"
    ]

Reply:
code incorrectly uses helper, so check has to fail, otherwise check is useless

Reply:
I don't know how to fix the incorrect use of helper, though...

Reply:
thats ok, not topic for this pr